app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestViewModel.java

```diff
@@ -0,0 +1,147 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.net.Uri;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.cmput301f20t21.bookfriends.entities.Book;
+import com.cmput301f20t21.bookfriends.entities.Request;
+import com.cmput301f20t21.bookfriends.repositories.BookRepository;
+import com.cmput301f20t21.bookfriends.repositories.RequestRepository;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.DocumentSnapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RequestViewModel extends ViewModel {
+    private MutableLiveData<Book> book;
+    private MutableLiveData<Uri> imageUri;
+    private MutableLiveData<List<Request>> requests;
+    private List<Request> requestsData;
+    private MutableLiveData<Integer> updatedPosition;
+
+    private final RequestRepository requestService = RequestRepository.getInstance();
+    private final BookRepository bookService = BookRepository.getInstance();
+
+    /**
+     * Function to get the book information from FireStore
+     * @param bookId we will query the book information based on the bookID
+     */
+    public void fetchBook(String bookId) {
+        bookService.getBookById(bookId).addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                DocumentSnapshot document = task.getResult();
+                if (document.exists()) {
+                    String imageName = (String) document.get("imageName");
```

Comment:
need a null check fro imageName here. I got an error on viewing requests of a book with no image.